### PR TITLE
Increase memory for node to fix master branch

### PR DESCRIPTION
**Pull request by fleupold**
Travis master is failing because it is currently running out of memory (cf. https://travis-ci.org/github/gnosis/dex-contracts/builds/662931725?utm_medium=notification&utm_source=github_status)

This PR increases the available RAM for our node process to 4GB which will unblock us. I will take a look at where we are using so much memory.

After increasing memory, it was still failing for testing the gas cap on our worst case scenario (reversion and reaplication of a solution with 30 trades). This looks to be related to the ganache version (it works on Ganache CLI v6.7.0 but fails on Ganache CLI v6.9.1 locally).

Travis is using @latest und thus may have started failing without us noticing via and explicit PR. I'm therefore making the dependency explicit inside our package.json (so that we will create a PR for updating it, which makes it easier to pinpoint failures due to version upgrades) and increase the gas limit to 6M in order to make the test pass.

### Test Plan

travis

### CODE REVIEW

**Review by anxolin — APPROVED** [review]
Since you added the dependency for ganache, could you add also a script to run it, and ideally something in the readme. (i.e. `yarn ganache`)

I think it's beneficial for example for avoiding users run into the gas limit problem. See https://gnosisinc.slack.com/archives/CPZA1AGKY/p1584354927002400

**Review by josojo — APPROVED** [review]
Thanks Felix,

the gas issue is really strange! Which ganache-cli version is closer to the current EVM spec?

Maybe it is due the fact that one ganache-cli was running with the old -before- Istanbul EVM, and now it switched to the new one, where the read opcode pricing was changed.

File: README.md [hunk 1]
```diff
@@ -34,6 +34,12 @@ npx truffle migrate --network $NETWORKNAME --reset
 yarn run networks-extract           # extracts deployed addresses to networks.json
 ```
 
+If you are building for a local development network, ganache has to be running locally. For this you can e.g. in a separate shell run
+
+```sh
+yarn run ganache # start a development network (blocking)
```
Comment (anxolin) [inline README.md:40]: You don't need the word `run`

**Comment by fleupold** [comment]
> Maybe it is due the fact that one ganache-cli was running with the old -before- Istanbul EVM

I believe you are right.

Our last successful build (https://travis-ci.org/github/gnosis/dex-contracts/builds/662931580?utm_medium=notification&utm_source=github_status) was running with 6.7.0

The next version is https://github.com/trufflesuite/ganache-cli/releases/tag/v6.8.1 which introduces Istanbul and this one is already failing.

File: package.json [hunk 2]
```diff
@@ -25,7 +25,8 @@
     "spread-orders": "npx truffle exec scripts/stablex/place_spread_orders.js",
     "ts": "yarn run ts-contracts && yarn run ts-check",
     "ts-check": "tsc src/index.d.ts",
-    "ts-contracts": "typechain --target web3-v1 --outDir build/types build/contracts/{BatchExchange,SnappAuction}.json"
+    "ts-contracts": "typechain --target web3-v1 --outDir build/types build/contracts/{BatchExchange,SnappAuction}.json",
+    "ganache": "ganache-cli -l 8e6"
```
Comment (anxolin) [inline package.json]: Can you make it deterministic and listen to all interfaces, this is useful for example for our TheGraph setup.

```suggestion
    "ganache": "ganache-cli -h 0.0.0.0 --gasLimit 8e6 --deterministic"
```

**Comment by fleupold** [comment]
Memory consumption looks non-leaky just high. The graph is showing the resident set size on beforeEach test using `console.log(process.memoryUsage())`.

Screen Shot 2020-03-17 at 15 01 51

I think keeping 4GB of RAM is ok.

File: test/stablex/stablex_large_example.js [hunk 3]
```diff
@@ -9,7 +9,7 @@ contract("BatchExchange", async accounts => {
   describe("Large Examples [ @skip-on-coverage ]", () => {
     it("ensures hard gas limit on largest possible ring trade ", async () => {
       const batchExchange = await setupGenericStableX(30)
-      const fivePointFiveMillion = 5500000
+      const limit = 6e8
```
Comment (bh2smith) [inline test/stablex/stablex_large_example.js:12]: I like the change here, but maybe we could have been a bit more explicit and said `gasCap` or `gasLimit`.

**Pull request merged**